### PR TITLE
docs: use eslint-doc-generator for rule documentation

### PR DESCRIPTION
**Pull request by G-Rath**
@bmish I'm sure I'm missing something obvious, but I can't for the life of me get the README table to be generated even though I'm using the exact same config as in `eslint-plugin-jest` - would you mind having a look at it for me?

### CODE REVIEW

**Comment by bmish** [comment]
@G-Rath darn, looks like a bug snuck in with `0.26.0`. Use `0.25.0` until it's fixed.

**Comment by G-Rath** [comment]
Ok cool so it's not just me being silly - I thought `eslint-plugin-jest` was using the latest version and that was working fine, but maybe not 🤷

**Comment by bmish** [comment]
Issue should be fixed in: https://github.com/bmish/eslint-doc-generator/releases/tag/v0.26.1

File: .eslint-doc-generatorrc.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/** @type {import('eslint-doc-generator/dist/lib/options').GenerateOptions} */
```
Comment (bmish) [inline .eslint-doc-generatorrc.js:1]: This won't work anymore and needs to be switched to the public type:

```suggestion
/** @type {import('eslint-doc-generator').GenerateOptions} */
```

File: README.md [hunk 2]
```diff
@@ -66,17 +66,20 @@ for installations requiring long-term consistency.
...
-<!-- begin base rules list -->
+<!-- begin auto-generated rules list -->
 
-| Rule                                                                             | Description                            | Fixable      |
-| -------------------------------------------------------------------------------- | -------------------------------------- | ------------ |
-| [prefer-to-be-array](docs/rules/prefer-to-be-array.md)                           | Suggest using `toBeArray()`            | ![fixable][] |
-| [prefer-to-be-false](docs/rules/prefer-to-be-false.md)                           | Suggest using `toBeFalse()`            | ![fixable][] |
-| [prefer-to-be-object](docs/rules/prefer-to-be-object.md)                         | Suggest using `toBeObject()`           | ![fixable][] |
-| [prefer-to-be-true](docs/rules/prefer-to-be-true.md)                             | Suggest using `toBeTrue()`             | ![fixable][] |
-| [prefer-to-have-been-called-once](docs/rules/prefer-to-have-been-called-once.md) | Suggest using `toHaveBeenCalledOnce()` | ![fixable][] |
```
Comment (bmish) [inline README.md:77]: The old fixable badge needs to be deleted still.

**Pull request merged**